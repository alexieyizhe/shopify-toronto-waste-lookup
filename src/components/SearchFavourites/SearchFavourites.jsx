import React from 'react';
import styled from 'styled-components';
import posed, { PoseGroup } from 'react-pose';

import { siteSubtitle, favsEmptyPlaceholder } from '../../utils/siteData';

const ComponentContainer = styled.div`
  height: auto;
  min-height: 20vh;

  padding: ${props => props.theme.styling.bodySpacing};

  background-color: ${props => `${props.theme.colors.secondary}15`}; // 8 digit hex code includes alpha value
`;


const SubHeader = styled.div`
  font-weight: 600;
  font-size: 2vw;
  color: ${props => props.theme.colors.secondary};

  margin-bottom: 1vw;
`;

const DisclaimerAnim = {
  enter: {
    opacity: 1,
    delay: 450
  },
  exit: { opacity: 0 }
}

const EmptyDisclaimer = styled(posed.div(DisclaimerAnim))`
  width: 100%;
  padding: ${props => props.theme.styling.bodySpacing};
  color: grey;
  text-align: center;
`;

// Component
export default ({ children }) => (
  <ComponentContainer>
    <SubHeader>{siteSubtitle}</SubHeader>
    <PoseGroup>
      {children.length > 0 ? children : <EmptyDisclaimer key="emptyDisclaimerFavs"><span>{favsEmptyPlaceholder}</span></EmptyDisclaimer>}
    </PoseGroup>
  </ComponentContainer>
);
