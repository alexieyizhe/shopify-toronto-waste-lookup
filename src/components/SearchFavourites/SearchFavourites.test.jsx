import React from 'react';
import renderer from 'react-test-renderer';
import SearchFavourites from './SearchFavourites';

/* eslint-disable no-undef */
it('renders correctly', () => {
  const tree = renderer.create(<SearchFavourites />).toJSON();
  expect(tree).toMatchSnapshot();
});
