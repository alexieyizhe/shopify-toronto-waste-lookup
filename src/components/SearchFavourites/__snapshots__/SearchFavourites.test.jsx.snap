// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  height: auto;
  min-height: 20vh;
  padding: 1vw;
  background-color: #23975e15;
}

.c1 {
  font-weight: 600;
  font-size: 2vw;
  color: #23975e;
  margin-bottom: 1vw;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Favourites
  </div>
  <div>
    test div
  </div>
  <a
    href="test thing thing"
  >
    test link
  </a>
  <div>
    <div>
      wogewog
      <span>
        wgwegww????!@$!@
      </span>
    </div>
  </div>
</div>
`;
